src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -88,7 +88,7 @@ object Parsers {
     def syntaxError(msg: String, offset: Int = in.offset): Unit =
       if (offset > lastErrorOffset) {
         syntaxError(msg, Position(offset))
-        lastErrorOffset = in.offset
+        lastErrorOffset = offset
```

Comment:
I've restored this change to make tests pass.

But it seems following line is also problematic, `DefDcl ::= DefSig `:' Type` may not have `=`.

https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/parsing/Parsers.scala#L1831

It seems `lastErrorOffset = in.offset` accidentally masked the error above. I'll investigate if there's indeed a problem here and make another PR if necessary.